client/src/main.ts

```diff
@@ -154,26 +154,32 @@ async function setupApp(): Promise<void> {
   }
 
   if (isElectron()) {
+    let isQuitDialogOpen: boolean = false;
+
     if ((await libparsec.getPlatform()) === Platform.Windows) {
       const mountpoint = await libparsec.getDefaultMountpointBaseDir();
       window.electronAPI.sendMountpointFolder(mountpoint);
     }
 
     window.electronAPI.receive('parsec-close-request', async (force: boolean = false) => {
-      let answer: Answer = Answer.Yes;
-      if (force === false) {
-        answer = await askQuestion('quit.title', 'quit.subtitle', {
-          yesText: 'quit.yes',
-          noText: 'quit.no',
-        });
-      }
-      if (answer === Answer.Yes) {
-        const devices = await getLoggedInDevices();
-        await injectionProvider.cleanAll();
-        for (const device of devices) {
-          await logout(device.handle);
+      if (!isQuitDialogOpen) {
```

Comment:
Consider leaving the function early in these cases:
```typescript
if (isQuitDialogOpen) {
  return;
}
```
You avoid an extra indentation which also makes the diff way easier to read (and potential conflicts resolution easier to solve).